py3status/modules/external_script.py

```diff
@@ -29,8 +29,7 @@
 
 @author frimdo ztracenastopa@centrum.cz
 """
-
```

Comment:
>  tested this module today. I think it shouldn't go in.
>
> Sometimes I receive exceptions due to the nature of scripting... grep picked up nothing this time... or curl when internet is down... etc can return error codes. 1,2, etc...

As far as I can tell the original module couldn't cope with non zero return codes `check_output` throws an error so this is not a regression.

So I think it's worth it.  The only problem is that a script like `ls ~` will fail where currently it works.  One solution would be a patch like 

```
diff --git a/py3status/modules/external_script.py b/py3status/modules/external_script.py
index 52f76a9..d6e2548 100644
--- a/py3status/modules/external_script.py
+++ b/py3status/modules/external_script.py
@@ -29,7 +29,6 @@ external_script {
 
 @author frimdo ztracenastopa@centrum.cz
 """
-import os
 
 
 class Py3status:
@@ -43,7 +42,7 @@ class Py3status:
 
     def external_script(self):
         if self.script_path:
-            return_value = self.py3.command_output(os.path.expanduser(self.script_path))
+            return_value = self.py3.command_output(self.script_path, shell=True)
             # this is a convenience cleanup code to avoid breaking i3bar which
             # does not support multi lines output
             if len(return_value.split('\n')) > 2:
diff --git a/py3status/py3.py b/py3status/py3.py
index 36da818..d5ed51e 100644
--- a/py3status/py3.py
+++ b/py3status/py3.py
@@ -685,7 +685,7 @@ class Py3:
             msg = "Command '{cmd}' {error}"
             raise Exception(msg.format(cmd=command[0], error=e))
 
-    def command_output(self, command):
+    def command_output(self, command, shell=False):
         """
         Run a command and return its output as unicode.
         The command can either be supplied as a sequence or string.
@@ -697,7 +697,7 @@ class Py3:
             command = shlex.split(command)
         try:
             process = Popen(command, stdout=PIPE, stderr=PIPE,
-                            universal_newlines=True)
+                            universal_newlines=True, shell=shell)
         except Exception as e:
             msg = "Command '{cmd}' {error}"
             raise Exception(msg.format(cmd=command[0], error=e))
```

I would also still prefer to slowly improve this module rather than adding #663 as  much of the stuff in that could be done via the formatter or in other 'universal' ways.  I'm planning to add some new formatter features soon as I've cleaned up the code there so that adding new features is easier see #752